src/session/mod.rs

```diff
@@ -356,6 +398,23 @@ impl Session {
         self_.user_list.get_or_init(|| UserList::new(self))
     }
 
+    pub fn basic_group_list(&self) -> &BasicGroupList {
+        let self_ = imp::Session::from_instance(self);
+        self_.basic_group_list.get_or_init(|| BasicGroupList::new())
+    }
+
+    pub fn supergroup_list(&self) -> &SupergroupList {
+        let self_ = imp::Session::from_instance(self);
+        self_.supergroup_list.get_or_init(|| SupergroupList::new())
```

Comment:
That Clippy again :-)
```
warning: redundant closure
   --> src/session/mod.rs:408:43
    |
408 |         self_.supergroup_list.get_or_init(|| SupergroupList::new())
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `SupergroupList::new`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_closure
```

Reply:
I really should start using it, sorry :p